clippy_dev/src/update_lints.rs

```diff
@@ -36,6 +36,57 @@ pub enum UpdateMode {
 pub fn update(update_mode: UpdateMode) {
     let (lints, deprecated_lints, renamed_lints) = gather_all();
     generate_lint_files(update_mode, &lints, &deprecated_lints, &renamed_lints);
+    remove_old_files(update_mode);
+}
+
+/// Remove files no longer needed after <https://github.com/rust-lang/rust-clippy/pull/9541>
+/// that may be reintroduced unintentionally
+fn remove_old_files(update_mode: UpdateMode) {
```

Comment:
Please add a fixme comment, that this should only exist temporarily, and should be removed once no PRs that add those files exist anymore